src/emulator/functionsEmulator.ts

```diff
@@ -122,143 +135,20 @@ export class FunctionsEmulator implements EmulatorInstance {
     // A trigger named "foo" needs to respond at "foo" as well as "foo/*" but not "fooBar".
     const httpsFunctionRoutes = [httpsFunctionRoute, `${httpsFunctionRoute}/*`];
 
-    const backgroundHandler = async (req: express.Request, res: express.Response) => {
-      const method = req.method;
-      const triggerId = req.params.trigger_name;
-
-      EmulatorLogger.log("DEBUG", `Accepted request ${method} ${req.url} --> ${triggerId}`);
-
-      const reqBody = (req as RequestWithRawBody).rawBody;
-      const proto = JSON.parse(reqBody.toString());
-
-      const worker = this.startFunctionRuntime(
-        bundleTemplate,
-        triggerId,
-        EmulatedTriggerType.BACKGROUND,
-        nodeBinary,
-        proto
-      );
-
-      worker.onLogs((el: EmulatorLog) => {
-        if (el.level === "FATAL") {
-          res.send(el.text);
-        }
-      });
-
-      // For analytics, track the invoked service
-      if (triggerId) {
-        const trigger = this.getTriggerById(triggerId);
-        track(EVENT_INVOKE, getFunctionService(trigger));
-      }
-
-      await worker.waitForDone();
-      return res.json({ status: "acknowledged" });
+    // TODO(samstern): I don't like needing to pass the bundle template and stuff in here
+    const backgroundHandler: express.RequestHandler = async (
+      req: express.Request,
+      res: express.Response
+    ) => {
+      this.handleBackgroundTrigger(req, res);
```

Comment:
These two handlers were previously inline in `createHubServer`.  It's much more readable to have them stand alone.